src/main/scala/scala/concurrent/stm/skel/HashTrieTMap.scala

```diff
@@ -68,6 +72,6 @@ private[skel] class HashTrieTMap[A, B] private (root0: Ref.View[TxnHashTrie.Node
   def put(key: A, value: B)(implicit txn: InTxn): Option[B] = txnPut(key, value)
   def remove(key: A)(implicit txn: InTxn): Option[B] = txnRemove(key)
 
-  def transform(f: (A, B) => B)(implicit txn: InTxn): this.type = { single transform f ; this }
-  def retain(p: (A, B) => Boolean)(implicit txn: InTxn): this.type = { single retain p ; this }
+  def transform(f: (A, B) => B)(implicit txn: InTxn): this.type = { single mapValuesInPlace f ; this }
+  def filterInPlace(p: ((A, B)) => Boolean)(implicit txn: InTxn): this.type = { single filterInPlace p ; this }
```

Comment:
We should use a `Function2` here as parameter, like `retain` used to take.

Reply:
https://static.javadoc.io/org.scala-lang/scala-library/2.13.0-M4/scala/collection/mutable/Map.html#filterInPlace(p:((K,V))=%3EBoolean):MapOps.this.type

I agree, let's be consistent with the bug in M4. I can do a follow-up PR with M5.